postgresql.go

```diff
@@ -180,32 +180,32 @@ func (p *postgresql) DumpSchema(w io.Writer) error {
 }
 
 func (p *postgresql) LoadSchema(r io.Reader) error {
-	cmd := exec.Command("psql", p.URL())
-	in, err := cmd.StdinPipe()
+	// Open DB connection on the target DB
+	deets := p.ConnectionDetails
+	db, err := sqlx.Open(deets.Dialect, p.MigrationURL())
 	if err != nil {
-		return err
+		return errors.WithMessage(err, fmt.Sprintf("unable to load schema for %s", deets.Database))
 	}
-	go func() {
-		defer in.Close()
-		io.Copy(in, r)
-	}()
-	Log(strings.Join(cmd.Args, " "))
-
-	bb := &bytes.Buffer{}
-	cmd.Stdout = bb
-	cmd.Stderr = bb
+	defer db.Close()
 
-	err = cmd.Start()
+	// Get reader contents
+	contents, err := ioutil.ReadAll(r)
 	if err != nil {
-		return errors.WithMessage(err, bb.String())
+		return err
+	}
+
+	if len(contents) == 0 {
+		fmt.Printf("schema is empty for %s, skipping\n", deets.Database)
+		return nil
 	}
 
-	err = cmd.Wait()
+	// From the sqlx package docs, this works with pq driver
+	_, err = db.Exec(string(contents))
```

Comment:
@stanislas-m nice :) this is very much needed. will there be any concern if the db have a lower db connection timeout? do we have db timeout added lately?

Reply:
I'm not sure about that, I'll take a look.

Reply:
@u007 I've checked, but there is no timeout management in pop right now.